### ISSUE TITLE
training - bbox loss is zero

### ISSUE THREAD

**Issue by VeeranjaneyuluToka** [mit-han-lab/bevfusion#371]
Hi kentang-mit,

I am trying to train a camera alone model with custom-dataset, i could see that training goes fine without any errors for 10 epochs. But i could not see train and validation loss graphs in tensorboard and bbox tensor values are zeros as shown below.

loss name: loss/object/heatmap/task0 tensor(0.1359, device='cuda:0', grad_fn=<MulBackward0>)
loss name: loss/object/bbox/task0 tensor(0., device='cuda:0', grad_fn=<MulBackward0>)
loss name: loss/object/heatmap/task1 tensor(3.8118, device='cuda:0', grad_fn=<MulBackward0>)
loss name: loss/object/bbox/task1 tensor(0.2242, device='cuda:0', grad_fn=<MulBackward0>)
loss name: loss/object/heatmap/task2 tensor(0.0070, device='cuda:0', grad_fn=<MulBackward0>)
loss name: loss/object/bbox/task2 tensor(0., device='cuda:0', grad_fn=<MulBackward0>)
loss name: loss/object/heatmap/task3 tensor(0.0021, device='cuda:0', grad_fn=<MulBackward0>)
loss name: loss/object/bbox/task3 tensor(0., device='cuda:0', grad_fn=<MulBackward0>)
loss name: loss/object/heatmap/task4 tensor(0.0054, device='cuda:0', grad_fn=<MulBackward0>)
loss name: loss/object/bbox/task4 tensor(0., device='cuda:0', grad_fn=<MulBackward0>)
loss name: loss/object/heatmap/task5 tensor(0.0071, device='cuda:0', grad_fn=<MulBackward0>)
loss name: loss/object/bbox/task5 tensor(0., device='cuda:0', grad_fn=<MulBackward0>)

Wondering what could be the reason for this behavior?

Thanks,
Veeru.

**Comment by kentang-mit** [comment]
One potential reason is that there is no matched box in the BEV feature map. According to the CenterPoint paper, the bbox regression loss is only computed on foreground objects. Therefore, you may want to check whether the target assigner worked properly.

**Comment by VeeranjaneyuluToka** [comment]
@kentang-mit , thanks for your reply, would you mind giving some pointers on how to check the target assigner? I am not sure if i followed you completely (still looking into CenterPiont paper and try to understand thier loss computation), would be good help if you can exapand a bit more here? Thanks!

**Comment by lcc815** [comment]
I have a same issue. and finally i find the reason is that my custom dataset class is not the same as these defined in configs/nuscenes/det/centerhead/default.yaml. Simply changing the classes will solve it. @VeeranjaneyuluToka

**Comment by VeeranjaneyuluToka** [comment]
@Icc815, I did not follow you completely, do you mean we need change class name in the config that you stated?

**Comment by lcc815** [comment]
@VeeranjaneyuluToka please refer to https://github.com/mit-han-lab/bevfusion/blob/main/configs/nuscenes/det/centerhead/default.yaml#L28

**Comment by VeeranjaneyuluToka** [comment]
@lcc815 , got it, i have this already but still i am getting the above loss values. BTW, are you getting tensorboard visualization?

**Comment by VeeranjaneyuluToka** [comment]
This is is because of GT shift which is due to transformations, i am able to correct it and will raise new issue if there is anything related to this. Thanks!
